sdks/java/core/src/main/java/org/apache/beam/sdk/util/IOChannelFactory.java

```diff
@@ -31,7 +32,10 @@
  *
  * <p>See <a href="http://docs.oracle.com/javase/7/docs/api/java/nio/channels/package-summary.html"
  * >Java NIO Channels</a>
+ *
+ * @deprecated This is under redesign, see: https://issues.apache.org/jira/browse/BEAM-59.
  */
+@Deprecated
```

Comment:
Experimental only means no compatibility guarantees, and sdk/util already suggests that.
I think we also want to discourage people from using it, since we are deprecating it right now.